### PR TITLE
polycap: new recipe

### PR DESCRIPTION
**Pull request by tschoonj**
This is initially for support on Linux and macOS, I will add Windows later when the feedstock is up and running, as I expect it will be time-consuming.

@PieterTack, can you please add a comment here saying you don't mind being a maintainer? Thanks!

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by tschoonj** [comment]
Any thoughts on how I can fix these packaging errors on Linux?

```
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libxrl.so.11 found in conda-forge::xraylib-4.0.0-py37hf476568_0
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libopenblas.so.0 found in conda-forge::libopenblas-0.3.7-h5ec1e0e_6
  ERROR (polycap,lib/libpolycap.so.0.0.0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libgsl.so.25 found in conda-forge::gsl-2.6-h294904e_0
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libhdf5.so.103 found in conda-forge::hdf5-1.10.5-nompi_h3c11f04_1104
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libgomp.so.1 found in conda-forge::_openmp_mutex-4.5-0_gnu
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h516909a_1006
  ERROR (polycap,lib/libpolycap.so.0.0.0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (polycap,lib/libpolycap.so.0.0.0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,lib/libpolycap.so.0.0.0): Needed DSO lib/libhdf5_hl.so.100 found in conda-forge::hdf5-1.10.5-nompi_h3c11f04_1104
  ERROR (polycap,lib/libpolycap.so.0.0.0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (polycap,lib/libpolycap.so.0.0.0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (polycap,bin/polycap): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,bin/polycap): lib/libpolycap.so.0 found in this package
  ERROR (polycap,bin/polycap): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,bin/polycap): Needed DSO lib/libgomp.so.1 found in conda-forge::_openmp_mutex-4.5-0_gnu
  ERROR (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libxrl.so.11 found in conda-forge::xraylib-4.0.0-py37hf476568_0
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libgsl.so.25 found in conda-forge::gsl-2.6-h294904e_0
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libhdf5.so.103 found in conda-forge::hdf5-1.10.5-nompi_h3c11f04_1104
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): lib/libpolycap.so.0 found in this package
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libhdf5_hl.so.100 found in conda-forge::hdf5-1.10.5-nompi_h3c11f04_1104
  ERROR (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libopenblas.so.0 found in conda-forge::libopenblas-0.3.7-h5ec1e0e_6
   INFO (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h516909a_1006
  ERROR (polycap,lib/python3.7/site-packages/polycap.cpython-37m-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 not found in any CDT/compiler package, nor the whitelist?!
WARNING (polycap): run-exports library package conda-forge::python-3.7.6-h8356626_5_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
   INFO (polycap): plugin library package conda-forge::numpy-1.18.5-py37h8960a57_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (polycap): run-exports library package conda-forge::libgomp-9.2.0-h24d8f2e_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

Thanks!

**Comment by PieterTack** [comment]
> 
> 
> @PieterTack, can you please add a comment here saying you don't mind being a maintainer? Thanks!
> 

I do not mind being a maintainer :)

**Comment by tschoonj** [comment]
@conda-forge/help-python-c Could someone have a look please? 

Thanks!

File: recipes/polycap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - {{ compiler('c') }}  # [unix]
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux]
+  host:
+    - python
+    - cython
+    - hdf5
+    - gsl
+    - numpy >=1.14
+    - pkg-config
```
Comment (scopatz) [inline recipes/polycap/meta.yaml:26]: pkg-config should be in build, not host

Comment (tschoonj) [reply recipes/polycap/meta.yaml]: @scopatz Thanks, that fixed it!

**Comment by tschoonj** [comment]
@conda-forge/staged-recipes would it be possible to merge this one in? Many thanks!

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**